valhalla/mjolnir/landmark_builder.h

```diff
@@ -0,0 +1,90 @@
+#pragma once
+
+#include <memory>
+#include <string>
+#include <vector>
+
+#include <boost/property_tree/ptree.hpp>
+#include <string>
+#include <unordered_map>
+
+namespace valhalla {
+namespace mjolnir {
+static const std::string default_landmark_name = "";
+
+// obvious landmark types for vehicle routing
+enum class LandmarkType : uint8_t {
+  restaurant = 1,
+  fast_food = 2,
+  cafe = 3,
+  fuel = 4,
+  bank = 5,
+  pharmacy = 6,
+  kindergarten = 7,
+  bar = 8,
+  hospital = 9,
+  post_office = 10,
+  pub = 11,
+  clinic = 12,
+  police = 13,
+  fire_station = 14,
+  car_wash = 15,
+  theatre = 16,
+  cinema = 17,
+  casino = 18,
+};
```

Comment:
i disagree that some of these will be obvious, in fact i think some of them require a name. id really like to stress that reading what i wrote in the initial issue #4037 regarding landmarks, specifically the visibility/verbosity/specificity sections. we aren't making a poi database we are trying to pick features that might stand out more than a road sign because at the decision point in the route you have a high pressure situation where you must decide quickly. if we are at all unsure about a specific amenity type then we should err on not using it. anyway here are some of my thoughts on this inline:

```suggestion
// obvious landmark types for vehicle routing
enum class LandmarkType : uint8_t {
  // these will almost always be obvious by their function and don't require a name
  fuel = 1,
  post_office = 2,
  police = 3,
  fire_station = 4,
  car_wash = 5,

  // could be obvious if its a chain but could also just be a building
  restaurant = 6,
  // this will always be a chain in which case the name is a required distinguisher (as they are often next to each other on either side of a street)
  fast_food = 7,
  // this could be a chain making it easier to distinguish (starbucks) but could be a one off cafe in which case the name will be required
  cafe = 8,
  // on the fence about this one, these are often obvious without a name, maybe should go in the upper list?
  bank = 10,
  // same with this one, in some countries these are obvious and have the little plus sign, in america its not the case and they can be different chains or could be part of a grocery store or anything... speaking of grocery store, why isnt that on the list? also why arent department stores on the list, surely stores would be a good landmark to use (again when they are named)
  pharmacy = 11,
  // again these could just be a random building with nothing other than a sign like "early years" or something, they better have a name or it will be ambiguous
  kindergarten = 12,
  // same thing, often just a building
  bar = 13,
  // this could be very very big like a college campus, i think we should remove this
  hospital = 14,
  // same as bar could be just a building need at least a name to have a chance of seeing it
  pub = 16,
  // generic, could be manyy many types of clinics also often there are multiple of these located in same "complex", you might have a dentist, orthodontist, ear nose throat and eye specialists all next to each other. 
  clinic = 17,
  // on the fence about this one, to me this would be in the same category as "museum" or something where its so specialized maybe it doesnt need a name. what keeps me from say ing that is you often cant tell from the outside what function these amenities serve until you read the name
  theatre = 16,
  // this could work without a name in my opinion, they often have a board out front that shows what movies are playing which would probably remove the need for a name
  cinema = 17,
  // these are usually quite large as well but not usually multiple buildings, how often are they located near one another though? if its often perhaps we need a name to differentiate between this casino and the next one down on the next block? maybe that argument is a good argument for everything has to have a name?
  casino = 18,
};
```

Reply:
I've added the remarks. I feel this is going to be a long term discussion and decision :)